### PR TITLE
config: disable auto-routing by default

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
- To make the default config more secure, disable  auto-routing by default.

**Checklist:**
- [x] Securely signed commits
- [] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [x] User guide updated
  - [x] Needs  #5758
  - [x] Tutorial
  - [x] Routing
  - [x] Controller
- [x] Conforms to style guide

### CODE REVIEW

File: app/Config/Routes.php [hunk 1]
```diff
@@ -21,7 +21,7 @@
 $routes->setDefaultMethod('index');
 $routes->setTranslateURIDashes(false);
 $routes->set404Override();
-$routes->setAutoRoute(true);
+$routes->setAutoRoute(false);
```
[the remark this answers is not in the document]
Comment (kenjis) [reply app/Config/Routes.php]: It is commented out in a later commit.
https://github.com/codeigniter4/CodeIgniter4/pull/5757/files#diff-a5a7e51bf35d7176e7dc1209e48e99d6b21af0a4dacb6c081d7394e156a38258R27

**Comment by MGatner** [comment]
To be clear I think this is a "policy decision" for us to make, but having community opinions could help inform that.

**Comment by sfadschm** [comment]
> This shouldn't affect existing projects so nothing is technically \"breaking\" but it could catch people off guard.

It only breaks if someone merges the app folder without reviewing the changes, right?

I am good with the changes. Auto-routing is convenient, but dangerous, so it should be a well-educated decision to enable it in a live project. With all the docs additions that kenjis pushed on this topic before, I think people have more than enough input on how to be lazy in a safe way ☺

**Comment by kenjis** [comment]
> It only breaks if someone merges the app folder without reviewing the changes, right?

Yes.

**Comment by kenjis** [comment]
@MGatner 

> Let's see what the rest of the team thinks, and any community input. Maybe a forum/Slack post is in order?

Do a poll on the forum?

**Comment by kenjis** [comment]
Rebased.

**Comment by MGatner** [comment]
Overwhelmingly positive response from https://forum.codeigniter.com/showthread.php?tid=81379 - most of the hesitancy seemed mostly around misunderstanding. Let's launch this 🚀

**Review by MGatner — APPROVED** [review]

**Comment by MGatner** [comment]
One afterthought: is this potentially affected by the Composer Autoload/Modules PR? I know Routes is one of the configurable modules.

**Comment by kenjis** [comment]
Rebased to fix conflicts.

**Comment by kenjis** [comment]
>  is this potentially affected by the Composer Autoload/Modules PR?

I think no.
This PR just disable auto-routing by default. 
Auto-routing only routes to controllers in `App\Controllers\`.
 Routes in modules are all defined routes.

**Pull request merged**